VideoLocker/src/main/java/org/edx/mobile/util/observer/AsyncCallableUtils.java

```diff
@@ -0,0 +1,37 @@
+package org.edx.mobile.util.observer;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+
+public enum AsyncCallableUtils {
+    ;
+
+    // Using AsyncTask.THREAD_POOL_EXECUTOR to automatically support Espresso idling resources
+    @VisibleForTesting
+    @NonNull
+    public static Executor EXECUTOR = AsyncTask.THREAD_POOL_EXECUTOR;
```

Comment:
The `AsyncTask.THREAD_POOL_EXECUTOR` has it's maximum pool size defined as `CPU_COUNT * 2 + 1` post-KitKat (and as 128 pre-KitKat). I think you should define your own `ThreadPoolExecutor` without a maximum limit. Note that OkHttp has a `Dispatcher` class that defines the total maximum requests and the maximum per host, and this is also used in the Retrofit 2 enqueueing logic, so we might also want to define a method that uses it.